### PR TITLE
Test examples of `use`

### PR DESCRIPTION
**Pull request by TrMen**
# Description
Add `test_examples()` to `use` that verifies its examples. For that, add the necessary definitions and the `PWD` to the `nu-cmd-lang` test support.

Note: `let-env` is a `nu-command` and thus not accessible from the `nu-cmd-lang` tests. So replace `let-env` with `$env.<var> =`. This should be fine, since `let-env` is supposed to be removed before 1.0 anyway.

# User-Facing Changes
Nothing

# Question for reviewer
Is there any particular reason why so many core commands (parser keywords) don't verify their examples with tests? E.g. `break`, `hide`, `overlay use`, etc. (I think most of them?). If there is no particular reason, I'd make a follow up PR to enable tests for as many of them as possible.

### CODE REVIEW

File: crates/nu-cmd-lang/src/example_test.rs [hunk 1]
```diff
@@ -61,19 +64,27 @@ mod test_examples {
             // Try to keep this working set small to keep tests running as fast as possible
             let mut working_set = StateWorkingSet::new(&engine_state);
             working_set.add_decl(Box::new(Break));
+            working_set.add_decl(Box::new(Collect));
```
Comment (fdncred) [inline crates/nu-cmd-lang/src/example_test.rs:67]: Are all these new `add_decl`'s needed for this PR? We try to keep these at a minimum and not allow every command to be tested in examples.

Comment (TrMen) [reply crates/nu-cmd-lang/src/example_test.rs:67]: I suppose part of the reason I opened this PR is that I'm wondering which examples you do and don't want to test. To test all the existing examples of `use`, yes all those `add_decl` are needed.

Comment (fdncred) [reply crates/nu-cmd-lang/src/example_test.rs:67]: Good question. I'm not sure exactly myself. Let's see what others say.

**Comment by amtoine** [comment]
@TrMen 
imo, the more we test, the better :+1: 

assuming we stay reasonable and do not add obviously useless or redundant tests, we want to have a rock-solid shell, so testing more core commands looks probably like a good step forward :relieved: 

could you fix the merge conflicts so that we can move forward? :yum:

**Comment by TrMen** [comment]
I fixed the conflicts

**Review by amtoine — APPROVED** [review]
imo these changes look good, thanks @TrMen :ok_hand:

**Comment by fdncred** [comment]
> Is there any particular reason why so many core commands (parser keywords) don't verify their examples with tests? E.g. break, hide, overlay use, etc. (I think most of them?). If there is no particular reason, I'd make a follow up PR to enable tests for as many of them as possible.

The more tests the less breakage usually. So, I'd vote for adding more tests.

**Comment by fdncred** [comment]
Thanks. I see no real harm in this. Sorry it took so long.

**Pull request merged**